src/ui/components/inputs/Checkbox/Checkbox.tsx

```diff
@@ -1,84 +1,210 @@
-      {label ? (
-        <LabelComponent>
-          {label}
-          {required ? '*' : null}
-        </LabelComponent>
-      ) : null}
+      {...focusProps}
+      {...hoverProps}>
+      <ContentContainer>
+        <LeftBox state={state} variant={variant}>
+          {isChecked ? <CheckboxMarkChecked /> : null}
+          {indeterminate ? <CheckboxMarkIndeterminate /> : null}
+        </LeftBox>
+        <RightBox>
+          <StyledBody disabled={disabled}>
+            {label}
+            {required ? '*' : null}
+          </StyledBody>
+          {description ? (
+            <StyledBodyAccentXs disabled={disabled}>{description}</StyledBodyAccentXs>
+          ) : null}
+        </RightBox>
+        {asset ? (
+          <BottomBox>
+            <CheckboxAsset {...asset} disable={disabled} />
+          </BottomBox>
+        ) : null}
+      </ContentContainer>
+      {collapsed ? <CollapsedContainer>{collapsed}</CollapsedContainer> : null}
     </CheckboxContainer>
   )
 }
 
-const CheckboxContainer = styled(TouchableOpacity)({
-  display: 'flex',
-  flexDirection: 'row',
+type ContainerProps = {
+  state: CheckboxState
+  variant: CheckboxVariant
+  collapsed?: React.ReactNode
+  display?: CheckboxDisplay
+  isHover?: boolean
+  isFocus?: boolean
+}
+
+const CheckboxContainer = styled(TouchableOpacity)<ContainerProps>(({
+  theme,
+  state,
+  variant,
+  display,
+  isFocus,
+  isHover,
+  collapsed,
+}) => {
+  const { borderColor, backgroundColor } = getCheckboxColors(
+    state,
+    variant,
+    !!collapsed,
+    theme,
+    'container'
+  )
+  const isDetailed = variant === 'detailed'
+  const isFilled = display === 'fill'
+  return {
+    cursor: 'pointer',
+    width: isFilled ? '100%' : 'fit-content',
+    ...(isDetailed && {
+      backgroundColor,
+      border: 1,
+      borderColor,
+      borderRadius: getSpacing(2),
+      padding: getSpacing(4),
+    }),
+    ...customFocusOutline({ isFocus }),
+    ...getHoverStyle(theme.designSystem.color.text.default, isHover),
+  }
+})
+
+const ContentContainer = styled.View({
   alignItems: 'center',
-  cursor: 'pointer',
+  flexDirection: 'row',
   columnGap: getSpacing(4),
 })
 
-const Box = styled.View<IsCheckedProps & { size: number }>(({ isChecked, theme }) => ({
-  width: theme.checkbox.size,
-  height: theme.checkbox.size,
-  justifyContent: 'center',
-  alignItems: 'center',
-  borderRadius: theme.borderRadius.checkbox,
-  border: theme.checkbox.border.size,
-  borderColor: isChecked
-    ? theme.designSystem.color.border.brandPrimary
-    : theme.designSystem.color.border.default,
-  backgroundColor: isChecked
-    ? theme.designSystem.color.background.brandPrimary
-    : theme.designSystem.color.background.default,
+type LeftBoxProps = {
+  variant: CheckboxVariant
+  state: CheckboxState
+}
+
+const LeftBox = styled.View<LeftBoxProps>(({ theme, variant, state }) => {
+  const { borderColor, backgroundColor } = getCheckboxColors(state, variant, false, theme, 'mark')
```

Comment:
je n'aime pas `false` en argument positionnel

en lisant uniquement la ligne 175, je ne comprends pas ce que signifie ce `false`

pour comprendre, je suis obligé d'aller à la définition de la fonction

je recommande de passer un objet en argument

```suggestion
  const { borderColor, backgroundColor } = getCheckboxColors({ state, variant, collapsed: false, theme, options: 'mark' })
```